### PR TITLE
SAMZA-1233: Create SystemAdmin only for JobCoordinator System in SamzaTaskProxy

### PR DESCRIPTION
**Pull request by shanthoosh**

### CODE REVIEW

**Comment by shanthoosh** [comment]
@jmakes Please take a look when you get a chance.

File: samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java [hunk 1]
```diff
@@ -155,8 +156,16 @@ protected JobModel getJobModel(JobInstance jobInstance) {
       changelogManager.start();
       LocalityManager localityManager = new LocalityManager(coordinatorSystemProducer, coordinatorSystemConsumer);
       localityManager.start();
-      return JobModelManager.readJobModel(config, changelogManager.readChangeLogPartitionMapping(), localityManager,
-                                          new StreamMetadataCache(JobModelManager.getSystemAdmins(config), 0, SystemClock.instance()), null);
+
+      String jobCoordinatorSystemName = config.get(JobConfig.JOB_COORDINATOR_SYSTEM());
+
+      /**
+       Select job coordinator system properties from config and instantiate SystemAdmin for it alone.
```
Comment (prateekm) [inline samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java]: Add '*' on each javadoc line.

File: samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java [hunk 2]
```diff
@@ -155,8 +156,16 @@ protected JobModel getJobModel(JobInstance jobInstance) {
...
+       Instantiating SystemAdmin's for other input/output systems defined in config is unnecessary.
+      */
+      Config systemAdminConfig = config.subset(String.format("systems.%s", jobCoordinatorSystemName), false);
+      StreamMetadataCache streamMetadataCache = new StreamMetadataCache(JobModelManager.getSystemAdmins(systemAdminConfig), 0, SystemClock.instance());
```
Comment (prateekm) [inline samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java]: Let's not inline side-effecty/meaty method calls in parameters. Same in next line for readChaneLogPartitionMapping.

File: samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java (hunk 1 above)
Comment (shanthoosh) [inline samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java]: Fixed.

File: samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java (hunk 2 above)
Comment (shanthoosh) [inline samza-rest/src/main/java/org/apache/samza/rest/proxy/task/SamzaTaskProxy.java]: Moved method calls into separate lines. For methods with side-effects this makes sense.

Just curious, reason for chaining is not create unnecessary temporary variables who scope go away after that line(making it more verbose).

Please share your thoughts.

**Pull request closed without merging**